src/main/java/de/hysky/skyblocker/skyblock/item/tooltip/info/TooltipInfoType.java

```diff
@@ -0,0 +1,59 @@
+	DataTooltipInfoType<Object2DoubleMap<String>> ONE_DAY_AVERAGE = ofData(itemTooltip -> itemTooltip.enableAvgBIN, "https://hysky.de/api/auctions/lowestbins/average/1day.json", CodecUtils.object2DoubleMapCodec(Codec.STRING), false, Object2DoubleMap::containsKey);
+	DataTooltipInfoType<Object2DoubleMap<String>> THREE_DAY_AVERAGE = ofData(itemTooltip -> itemTooltip.enableAvgBIN, itemTooltip -> itemTooltip.enableAvgBIN || getConfig().uiAndVisuals.searchOverlay.enableAuctionHouse, "https://hysky.de/api/auctions/lowestbins/average/3day.json", CodecUtils.object2DoubleMapCodec(Codec.STRING), false, Object2DoubleMap::containsKey);
+	DataTooltipInfoType<Object2IntMap<String>> MOTES = ofData(itemTooltip -> itemTooltip.enableMotesPrice, itemTooltip -> itemTooltip.enableMotesPrice && Utils.isInTheRift(), "https://hysky.de/api/motesprice", CodecUtils.object2IntMapCodec(Codec.STRING), true, Object2IntMap::containsKey);
+	TooltipInfoType OBTAINED = ofSimple(itemTooltip -> itemTooltip.enableObtainedDate);
+	DataTooltipInfoType<Map<String, String>> MUSEUM = ofData(itemTooltip -> itemTooltip.enableMuseumInfo, "https://hysky.de/api/museum", Codec.unboundedMap(Codec.STRING, Codec.STRING), true, Map::containsKey);
+	DataTooltipInfoType<Map<String, String>> COLOR = ofData(itemTooltip -> itemTooltip.enableExoticTooltip, "https://hysky.de/api/color", Codec.unboundedMap(Codec.STRING, Codec.STRING), true, Map::containsKey);
+	DataTooltipInfoType<Map<String, Accessory>> ACCESSORIES = ofData(itemTooltip -> itemTooltip.enableAccessoriesHelper, "https://hysky.de/api/accessories", Accessory.MAP_CODEC, true, Map::containsKey, AccessoriesHelper::refreshData);
+	TooltipInfoType ESTIMATED_ITEM_VALUE = ofSimple(itemTooltip -> itemTooltip.enableEstimatedItemValue);
+
+	/**
+	 * @return whether the tooltip should be shown
+	 */
+	boolean isTooltipEnabled();
+
+	private static SkyblockerConfig getConfig() {
+		return SkyblockerConfigManager.get();
+	}
+
+	private static TooltipInfoType ofSimple(Predicate<GeneralConfig.ItemTooltip> tooltipEnabled) {
+		return new SimpleTooltipInfo(tooltipEnabled);
+	}
+
+	@SafeVarargs
+	private static <T> DataTooltipInfoType<T> ofData(Predicate<GeneralConfig.ItemTooltip> tooltipEnabled, String address, Codec<T> codec, boolean cacheable, BiPredicate<T, String> contains, Consumer<T>... callbacks) {
+		return ofData(tooltipEnabled, tooltipEnabled, address, codec, cacheable, contains, callbacks);
+	}
+
+	@SafeVarargs
+	private static <T> DataTooltipInfoType<T> ofData(Predicate<GeneralConfig.ItemTooltip> tooltipEnabled, Predicate<GeneralConfig.ItemTooltip> dataEnabled, String address, Codec<T> codec, boolean cacheable, BiPredicate<T, String> contains, Consumer<T>... callbacks) {
```

Comment:
These calls are a bit unreadable. I would have put `dataEnabled` and `tooltipEnabled` last.